cogs/mailinglist.py

```diff
@@ -0,0 +1,185 @@
+from discord.ext import commands
+import discord
+import aiohttp
+import json
+import os
+import datetime
+from dateutil import parser
+import asyncio
+import re
+import traceback
+from .utils import checks
+import logging
+
+"""
+Schema of Subs
+[
+	"subs": {
+		"listName": [1112, 32],
+		"listName2": [123, 453]
+	},
+	"users": {
+		someUserId: ["listName", listName2"]
+	}
+]
+"""
+
+class Danbooru:
+    """
+    Danbooru requests and subscription service.
+    """
+    def __init__(self, bot):
+        self.bot = bot
+        self.session = aiohttp.ClientSession()
+		
+		# Make sure directory to store json is good, and then read a file if it exists
+        self.init_directories()
+		self.jsonDir = 'data/mailinglist/data.json'
+		with open(self.jsonDir, 'r') as f:
+			self.json = json.load(f)
+
+    def __unload(self):
+		return
+
+    def init_directories(self):
+        if not os.path.exists('data/mailinglist'):
+            os.mkdir('data/mailinglist')
+        if not os.path.exists('data/mailinglist/subs/'):
+            os.mkdir('data/mailinglist/subs')
+		if not os.file.exists('data/mailinglist/subs/data.json/'):
+		    newJson = open('data.json','w')
+			newJson.write('[]');
+			file.close()
+			
+	def writeUpdatedList(self):
+		try:
+			with open(self.jsonDir, 'w') as f:
+				json.dump(self.json, f)
+		except Exception as e:
+		   await self.bot.say('Error while persisting sub change: `{}`'.format(repr(e)))
```

Comment:
```python
def writeUpdatedList(self)
```

needs to be async too or else you can't use await, because method is not asynchronous